### ISSUE TITLE
Page router doesn't mount components. Refresh needed to properly mount.

### ISSUE THREAD

**Issue by korabdoroci** [nuejs/nue#263]
Minimum reproducible scenario:

Index.html:
![image](https://github.com/nuejs/nue/assets/41199833/681ed7cd-e22b-4bd3-a47e-0b6ecf6cb62b)

Site.yaml: (when router is false it works as intended)
![image](https://github.com/nuejs/nue/assets/41199833/e5abcc2a-b01b-40c4-9951-200919fcbbe2)

Page index.html:
![image](https://github.com/nuejs/nue/assets/41199833/de167aa8-879a-4873-8721-0d1448a81ef3)

Component.nue:
![image](https://github.com/nuejs/nue/assets/41199833/91db3b8f-f9e4-4bb0-be8f-27087bb4cb5a)

Before clicking the button:
![image](https://github.com/nuejs/nue/assets/41199833/6ef39ea6-600a-4532-8caa-13c891dd41ce)

After clicking the button:
![image](https://github.com/nuejs/nue/assets/41199833/a4903712-7ed5-446b-8151-73cd0a19902c)

After refreshing:
![image](https://github.com/nuejs/nue/assets/41199833/e0214fd4-5921-40a3-b0e3-6b4548316ec2)

**Comment by tipiirai** [comment]
Thanks! Need to check this out

**Comment by mszmida** [comment]
@Korab-Doroci The Nue layout system is based on `<header>`, `<main>` and `<footer>` tags. Page router is searching for these tags to replace its content.

In your example code should look like this:
```html
// index.html
<main>
  <a href="/page">go to page</a>
</main>
```

```html
// page/index.html
<main>
  <component />
</main>
```

However the page router indeed **does not** mount components correctly. I prepared a PR with fix for this functionality in #271.
